wasmCloud/wadm
# feat(multitenant)! Account prefixed API topics and manifest storage

## Feature or Problem
This PR ensures we are storing and querying manifests based on an account prefix if running in multitenant mode. This ensures that accounts can only access and modify manifests that are specifically bound to that account, rather than simply using a global lattice ID.

Due to possible edge cases, it's still recommended to keep lattice prefixes globally unique between accounts, as there are not e2e tests at this time to verify that functionality works without issues. This PR at least implements the portion of that that constrains manifests from conflicting between accounts.

## Related Issues
#132

## Release Information
v0.5.0

## Consumer Impact
Multitenancy hasn't released yet so this shouldn't affect consumers, however they will need to examine the account prefixed topic if manually looking up manifests in NATS.

## Testing



Built on platform(s)


Tested on platform(s)

### Unit Test(s)
Modified a few unit tests that were hitting deprecation warnings

### Acceptance or Integration
Will have to modify the e2e tests to properly import/export account prefixes

### Manual Verification
Using the multitenant docker compose, I was able to verify that if I put a manifest using one account, I cannot query that manifest in another account even if I use the correct lattice prefix 😄
